In [2]:
! pip install kaggle

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [5]:
! mkdir ~/.kaggle

mkdir: cannot create directory ‘/root/.kaggle’: File exists


In [9]:
! cp kaggle.json ~/.kaggle/
! chmod 600 ~/.kaggle/kaggle.json

In [10]:
! kaggle datasets download mukhazarahmad/world-population-data

  0% 0.00/69.6k [00:00<?, ?B/s]
100% 69.6k/69.6k [00:00<00:00, 38.3MB/s]


In [11]:
! unzip world-population-data.zip

Archive:  world-population-data.zip
  inflating: Population Data.csv     


In [12]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [13]:
df = pd.read_csv('Population Data.csv')
df.head()

Country Name Country Code     Indicator Name Indicator Code  \
0                        Aruba          ABW  Population, total    SP.POP.TOTL   
1  Africa Eastern and Southern          AFE  Population, total    SP.POP.TOTL   
2                  Afghanistan          AFG  Population, total    SP.POP.TOTL   
3   Africa Western and Central          AFW  Population, total    SP.POP.TOTL   
4                       Angola          AGO  Population, total    SP.POP.TOTL   

          1960         1961         1962         1963         1964  \
0      54608.0      55811.0      56682.0      57475.0      58178.0   
1  130692579.0  134169237.0  137835590.0  141630546.0  145605995.0   
2    8622466.0    8790140.0    8969047.0    9157465.0    9355514.0   
3   97256290.0   99314028.0  101445032.0  103667517.0  105959979.0   
4    5357195.0    5441333.0    5521400.0    5599827.0    5673199.0   

          1965  ...         2012         2013         2014         2015  \
0      58782.0  ...     102112.0     102880.0     103594.0     104257.0   
1  149742351.0  ...  552530654.0  567891875.0  583650827.0  600008150.0   
2    9565147.0  ...   30466479.0   31541209.0   32716210.0   33753499.0   
3  108336203.0  ...  376797999.0  387204553.0  397855507.0  408690375.0   
4    5736582.0  ...   25188292.0   26147002.0   27128337.0   28127721.0   

          2016         2017         2018         2019         2020  \
0     104874.0     105439.0     105962.0     106442.0     106585.0   
1  616377331.0  632746296.0  649756874.0  667242712.0  685112705.0   
2   34636207.0   35643418.0   36686784.0   37769499.0   38972230.0   
3  419778384.0  431138704.0  442646825.0  454306063.0  466189102.0   
4   29154746.0   30208628.0   31273533.0   32353588.0   33428486.0   

          2021  
0     106537.0  
1  702976832.0  
2   40099462.0  
3  478185907.0  
4   34503774.0  

[5 rows x 66 columns]

In [14]:
df.shape

(266, 66)

In [15]:
df.columns

Index(['Country Name', 'Country Code', 'Indicator Name', 'Indicator Code',
       '1960', '1961', '1962', '1963', '1964', '1965', '1966', '1967', '1968',
       '1969', '1970', '1971', '1972', '1973', '1974', '1975', '1976', '1977',
       '1978', '1979', '1980', '1981', '1982', '1983', '1984', '1985', '1986',
       '1987', '1988', '1989', '1990', '1991', '1992', '1993', '1994', '1995',
       '1996', '1997', '1998', '1999', '2000', '2001', '2002', '2003', '2004',
       '2005', '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013',
       '2014', '2015', '2016', '2017', '2018', '2019', '2020', '2021'],
      dtype='object')

**Analysis**:
*   So, from the dataset, we can understand that there are 66 columns and 266 rows.
*  62 columns were the years from 1960 to 2021 and the balance 4 columns were the Country Name, Country Code, Indicator Name, Indicator Code.

*   Country Name and Country Code are unique values (266 values) whereas the Indicator name and Indicator code are the same for all 266 rows.

In [16]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 266 entries, 0 to 265
Data columns (total 66 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Country Name    266 non-null    object 
 1   Country Code    266 non-null    object 
 2   Indicator Name  266 non-null    object 
 3   Indicator Code  266 non-null    object 
 4   1960            264 non-null    float64
 5   1961            264 non-null    float64
 6   1962            264 non-null    float64
 7   1963            264 non-null    float64
 8   1964            264 non-null    float64
 9   1965            264 non-null    float64
 10  1966            264 non-null    float64
 11  1967            264 non-null    float64
 12  1968            264 non-null    float64
 13  1969            264 non-null    float64
 14  1970            264 non-null    float64
 15  1971            264 non-null    float64
 16  1972            264 non-null    float64
 17  1973            264 non-null    flo

=> Since the Indicator Name and Code are not that much useful, we can drop those 2 columns



In [17]:
df = df.drop(['Indicator Name', 'Indicator Code'], axis=1)
df.head()

Country Name Country Code         1960         1961  \
0                        Aruba          ABW      54608.0      55811.0   
1  Africa Eastern and Southern          AFE  130692579.0  134169237.0   
2                  Afghanistan          AFG    8622466.0    8790140.0   
3   Africa Western and Central          AFW   97256290.0   99314028.0   
4                       Angola          AGO    5357195.0    5441333.0   

          1962         1963         1964         1965         1966  \
0      56682.0      57475.0      58178.0      58782.0      59291.0   
1  137835590.0  141630546.0  145605995.0  149742351.0  153955516.0   
2    8969047.0    9157465.0    9355514.0    9565147.0    9783147.0   
3  101445032.0  103667517.0  105959979.0  108336203.0  110798486.0   
4    5521400.0    5599827.0    5673199.0    5736582.0    5787044.0   

          1967  ...         2012         2013         2014         2015  \
0      59522.0  ...     102112.0     102880.0     103594.0     104257.0   
1  158313235.0  ...  552530654.0  567891875.0  583650827.0  600008150.0   
2   10010030.0  ...   30466479.0   31541209.0   32716210.0   33753499.0   
3  113319950.0  ...  376797999.0  387204553.0  397855507.0  408690375.0   
4    5827503.0  ...   25188292.0   26147002.0   27128337.0   28127721.0   

          2016         2017         2018         2019         2020  \
0     104874.0     105439.0     105962.0     106442.0     106585.0   
1  616377331.0  632746296.0  649756874.0  667242712.0  685112705.0   
2   34636207.0   35643418.0   36686784.0   37769499.0   38972230.0   
3  419778384.0  431138704.0  442646825.0  454306063.0  466189102.0   
4   29154746.0   30208628.0   31273533.0   32353588.0   33428486.0   

          2021  
0     106537.0  
1  702976832.0  
2   40099462.0  
3  478185907.0  
4   34503774.0  

[5 rows x 64 columns]

Analyse the 'Country Name' column

In [18]:
df['Country Name'].unique()

array(['Aruba', 'Africa Eastern and Southern', 'Afghanistan',
       'Africa Western and Central', 'Angola', 'Albania', 'Andorra',
       'Arab World', 'United Arab Emirates', 'Argentina', 'Armenia',
       'American Samoa', 'Antigua and Barbuda', 'Australia', 'Austria',
       'Azerbaijan', 'Burundi', 'Belgium', 'Benin', 'Burkina Faso',
       'Bangladesh', 'Bulgaria', 'Bahrain', 'Bahamas, The',
       'Bosnia and Herzegovina', 'Belarus', 'Belize', 'Bermuda',
       'Bolivia', 'Brazil', 'Barbados', 'Brunei Darussalam', 'Bhutan',
       'Botswana', 'Central African Republic', 'Canada',
       'Central Europe and the Baltics', 'Switzerland', 'Channel Islands',
       'Chile', 'China', "Cote d'Ivoire", 'Cameroon', 'Congo, Dem. Rep.',
       'Congo, Rep.', 'Colombia', 'Comoros', 'Cabo Verde', 'Costa Rica',
       'Caribbean small states', 'Cuba', 'Curacao', 'Cayman Islands',
       'Cyprus', 'Czechia', 'Germany', 'Djibouti', 'Dominica', 'Denmark',
       'Dominican Republic', 'Algeria',
 



*   There are no NULL values in the CountryName column

